hr_timesheet_task_domain/models/account_analytic_line.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2016 Tecnativa - Sergio Teruel
+# Copyright 2016-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    @api.onchange('project_id')
+    def onchange_project_id(self):
+        Module = self.env['ir.module.module']
+
+        project_stage_closed = Module.sudo().search([
```

Comment:
@alexey-pelykh IMO this addon we shoulld split this addons and create a clue one with the dependency on `project_stage_closed`.

Reply:
at least you could optimize the code by avoiding 1 SQL query on each call to ` onchange_project_id

```python

# check if project_stage_closed is installed
project_stage_closed = 'closed' in self.env['project.stage']._fields
```

Reply:
Thanks for the suggestion, adjusted the code